File: GTP2.5/gtp7.cpp
#include<iostream>

using namespace std;

void input(string num[], int pass[], bool state[], int SIZE) {
        for(int i = 0; i < SIZE; i++) {
                cout << "\nDatos del vuelo " << i+1 << ": \n";

                cout << "Número de vuelo: ";
                cin >> num[i];

                cout << "Cantidad de pasajeros: ";
                cin >> pass[i];

                cout << "Estado del vuelo: ";
                cin >> state[i];

                cout << "\n--------------------------\n";
        }
}

int main() {
        int N, pass_count = 0, fly_count = 0;
        float porc = 0;

        string num[100];
        int pass[100];
        bool state[100];

        cout << "Ingrese la cantidad de vuelos del día: ";
        cin >> N;

        input(num, pass, state, N);

        for(int i = 0; i < N; i++) {
                fly_count += (!state[i]);
                porc += (!state[i])*100.0/N;
                pass_count += pass[i];
        }

        cout << "La cantidad de vuelos retrasados son: " << fly_count;
        cout << "\nEl porcentaje de vuelos atrasados con respecto al total es de: " << porc;
        cout << "\nLa cantidad de pasajeros totales es: " << pass_count << "\n";

        return 0;
}

File: GTP2.5/gtp7.py
import random as r

def generar_n_vuelo(n_vuelo:list[str]):
        for i in range(len(n_vuelo)):
                for j in range(5):
                        n_vuelo[i] += f"{int(10 * r.random())}"

def estan_retrasados(horarios:list[bool]) -> int:
        contador = 0

        for i in range(len(horarios)):
                contador += not horarios[i]

        return contador        

def contar_pasajeros(pasajeros:list[int]) -> int:
        contador = 0

        for i in range(len(pasajeros)):
                contador += pasajeros[i]

        return contador

def main():
        cant_vuelos = int(input("Ingrese la cantidad de vuelos: "))
        
        # n_vuelo = [""] * cant_vuelos
        # generar_n_vuelo(n_vuelo)
        
        # n_vuelo = [f"{10000 + int(89999 * r.random())}" for n in range(cant_vuelos)]
        n_vuelo = [f"{(int(99999 * r.random())):05d}" for n in range(cant_vuelos)]
        pasajeros = [int(300* r.random()) for n in range(cant_vuelos)]
        en_horario = [r.choice((True, False)) for n in range(cant_vuelos)]

        print(n_vuelo)
        print(pasajeros)
        print(en_horario)

        cant_retrasados = estan_retrasados(en_horario)

        print(f"La cantidad de vuelos retrasados es: {cant_retrasados}")
        print(f"El porcentaje de vuelos atrasados con respecto al total es de: {(cant_retrasados * 100/cant_vuelos):.2f}%")
        print(f"La cantidad de pasajeros totales es: {contar_pasajeros(pasajeros)}")

if __name__ == '__main__':
        main()
